### PR TITLE
changes Custom selectlist component and file extensions

### PR DESCRIPTION
**Pull request by visha12l**

### CODE REVIEW

File: src/javascript/components/selectlistApp/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    openSelectList () {
+        this.setState({
+            selectListStatus: true,
+        });
+    }
+
+    closeSelectList () {
+        this.setState({
+            selectListStatus: !this.state.selectListStatus,
```
Comment (Shrikant9) [inline src/javascript/components/selectlistApp/index.js]: function name says **close**SelectList, although this assignment seems to toggle? You can change this to `selectListStatus: false`. 
But rather replace both functions with `toggleSelectList` that does this.

File: src/javascript/components/shared/selectList.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+   componentWillMount () {
+      document.addEventListener('click',this.closeSelectList);
+   }
+
+   componentWillUnmount () {
+      document.removeEventListener('click',this.closeSelectList);
+   }
+
+   openSelectList () {
```
Comment (Shrikant9) [inline src/javascript/components/shared/selectList.js]: You can apply the same logic here and replace `open` and `close` functions with just `toggle`.

Comment (visha12l) [reply src/javascript/components/shared/selectList.js]: closeSelectlist have other code except toggling the state

File: src/javascript/components/Home.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+export default class Home extends React.Component {
+  constructor(props) {
```
Comment (avaneeshtripathi) [inline src/javascript/components/Home.js]: You actually dont need a constructor for components which dont have states. Also read about stateless/functional components. There is no need to create a class if component doesnt have an state, we can use normal functions. Its good to go, but from coding point of view.

File: src/javascript/components/selectlistApp/index.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    render() {
+        return (
+            <div className="container">
+                <div className="topwrap clearfix">
+                    <h1 className="pull-left">Select List  App!</h1>
+                    <Link className="pull-right" to='/'>go to Home Page</Link>
+                </div>
+                <div>
+                    <a onClick={this.toggleSelectList.bind(this)}>open select list Component</a>
```
Comment (avaneeshtripathi) [inline src/javascript/components/selectlistApp/index.js]: no need to bind here as we already did that in constructor.

File: src/javascript/components/shared/selectList.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    render() {
+        //need to implement autocomplete functionality for showing countryList
+        let countryListArray = this.props.listData;
+        return (
+            <div className="App">
+                <div className="App-header clearfix">
+                    <h2 className="pull-left">{this.props.selectListName} SELECT DROPDWON</h2>
+                    <a  className="pull-right" onClick={this.closeComponent.bind(this)}>close {this.props.selectListName} Component</a>
```
Comment (avaneeshtripathi) [inline src/javascript/components/shared/selectList.js]: Bind all the functions in constructor. Code looks cleaner and easily understandable to other person.

File: src/javascript/components/shared/selectList.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                </div>
+                <div className="selectListWrap">
+                    <div ref='selectionarea' className='selectionarea'>
+                        <input
+                            type="text"
+                            ref="countryInput"
+                            placeholder={this.props.placeholder}
+                            onFocus={this.openSelectDropdown.bind(this)}
```
Comment (avaneeshtripathi) [inline src/javascript/components/shared/selectList.js]: here as well the binding.

**Comment by avaneeshtripathi** [comment]
everything else looks good.

File: src/javascript/components/shared/selectList.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+                        />
+                        {this.state.selectListDropdown &&
+                            <ul className='contryListing'>
+                                {underscore.map(countryListArray, (list, key) =>
+                                    <li key={key} onClick={this.selectValue.bind(this, list.name)}>{list.name}</li>)
```
Comment (visha12l) [inline src/javascript/components/shared/selectList.js]: @avaneeshtripathi  for this case, we  should not bind it in constructor , is it correct ???

Comment (avaneeshtripathi) [reply src/javascript/components/shared/selectList.js]: here we need to pass something to that function and we need to use call/apply/bind while using that. So, its fine for this case. But if we want, we can bind it in constructor as well it wont have any impact.

**Pull request merged**